File: Polimorfismo/Ejercicio1/Videojuego.py
class Videojuego:
    def __init__(self, nombre="", plataforma="", cantidadJugadores=0):
        self.nombre = nombre
        self.plataforma = plataforma
        self.cantidadJugadores = cantidadJugadores

    @classmethod
    def constructor1(cls, nombre, plataforma):
        return cls(nombre, plataforma, 0)

    @classmethod
    def constructor2(cls, nombre, plataforma, cantidadJugadores):
        return cls(nombre, plataforma, cantidadJugadores)
 
    def mostrar(self):
        print(f"Nombre: {self.nombre}")
        print(f"Plataforma: {self.plataforma}")
        print(f"Cantidad de jugadores: {self.cantidadJugadores}")

    
    def agregarUnJugador(self):
        self.cantidadJugadores += 1

    def agregarJugadores(self, cantidad):
        self.cantidadJugadores += cantidad


videojuego1 = Videojuego.constructor1("Minecraft", "PC")
videojuego2 = Videojuego.constructor2("FIFA 25", "PlayStation", 4)
videojuego3 = Videojuego.constructor2("Call of Duty", "Xbox", 2)


print("Estado inicial:")
videojuego1.mostrar()
videojuego2.mostrar()
videojuego3.mostrar()


videojuego1.agregarUnJugador()  
videojuego2.agregarJugadores(2)  
videojuego3.agregarUnJugador()  

print("\nDespues de agregar jugadores:")
videojuego1.mostrar()
videojuego2.mostrar()
videojuego3.mostrar()

File: Polimorfismo/Ejercicio1/Videojuego.java
package Polimorfismo.Ejercicio1;

public class Videojuego {
    private String nombre;
    private String plataforma;
    private int cantidadJugadores;

    public Videojuego() {
        this.nombre = "";
        this.plataforma = "";
        this.cantidadJugadores = 0;
    }

    public Videojuego(String nombre, String plataforma) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = 0;
    }

    public Videojuego(String nombre, String plataforma, int cantidadJugadores) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = cantidadJugadores;
    }

    public void mostrar() {
        System.out.println("Nombre: " + nombre);
        System.out.println("Plataforma: " + plataforma);
        System.out.println("Cantidad de jugadores: " + cantidadJugadores);
    }

    public void agregarUnJugador() {
        cantidadJugadores += 1;
    }

    public void agregarJugadores(int cantidad) {
        cantidadJugadores += cantidad;
    }

    // Main 
    public static void main(String[] args) {

        Videojuego videojuego1 = new Videojuego("Minecraft", "PC");
        Videojuego videojuego2 = new Videojuego("FIFA 25", "PlayStation", 2);
        Videojuego videojuego3 = new Videojuego("Call of Duty", "Xbox", 2);

        System.out.println("Estado inicial:");
        videojuego1.mostrar();
        videojuego2.mostrar();
        videojuego3.mostrar();

  
        videojuego1.agregarUnJugador(); 
        videojuego2.agregarJugadores(2); 
        videojuego3.agregarUnJugador(); 

        System.out.println("\nDespues de agregar jugadores:");
        videojuego1.mostrar();
        videojuego2.mostrar();
        videojuego3.mostrar();
    }
}

